fathom/src/core.rs

```diff
@@ -422,6 +431,88 @@ pub enum Const {
     Ref(u64),
 }
 
+impl PartialEq for Const {
+    fn eq(&self, other: &Self) -> bool {
+        match (*self, *other) {
+            (Const::Bool(a), Const::Bool(b)) => a == b,
+            (Const::U8(a, _), Const::U8(b, _)) => a == b,
+            (Const::U16(a, _), Const::U16(b, _)) => a == b,
+            (Const::U32(a, _), Const::U32(b, _)) => a == b,
+            (Const::U64(a, _), Const::U64(b, _)) => a == b,
+            (Const::S8(a), Const::S8(b)) => a == b,
+            (Const::S16(a), Const::S16(b)) => a == b,
+            (Const::S32(a), Const::S32(b)) => a == b,
+            (Const::S64(a), Const::S64(b)) => a == b,
+            (Const::F32(a), Const::F32(b)) => a == b,
+            (Const::F64(a), Const::F64(b)) => a == b,
+            (Const::Pos(a), Const::Pos(b)) => a == b,
+            (Const::Ref(a), Const::Ref(b)) => a == b,
+            _ => false,
+        }
+    }
+}
+
+pub trait ToBeBytes<const N: usize> {
+    fn to_be_bytes(self) -> [u8; N];
+}
+
+macro_rules! impl_styled_int {
+    ($($ty:ty),*) => {
+        $(
+        impl ToBeBytes<{std::mem::size_of::<$ty>()}> for &$ty {
+            fn to_be_bytes(self) -> [u8; std::mem::size_of::<$ty>()] {
+                <$ty>::to_be_bytes(*self)
+            }
+        }
+
+        impl Styled<{std::mem::size_of::<$ty>()}> for &$ty {}
+        )*
+    };
+}
+
+impl_styled_int!(u8, u16, u32, u64);
+
+pub trait Styled<const N: usize>:
+    std::fmt::Display + Copy + std::fmt::LowerHex + std::fmt::Binary + ToBeBytes<N>
+{
+}
+
+impl IntStyle {
+    pub fn format<T: Styled<N>, const N: usize>(&self, number: T) -> String {
+        match self {
+            IntStyle::Binary => format! {"0b{:b}", number},
```

Comment:
Why do we use braces for some of these `format!` invocations? Is that for some sort of escaping reason?

Reply:
I suspect I just mistyped `(` and being a macro it still works. It wasn't intentional, will fix.